src/compiler/scala/tools/nsc/ast/parser/Parsers.scala

```diff
@@ -829,11 +829,14 @@ self =>
         s"Incompatible args to makeBinop: !isExpr but targs=$targs")
 
       def mkSelection(t: Tree) = {
-        val pos = opPos union t.pos
+        val pos = (opPos | t.pos) makeTransparentIf !treeInfo.isLeftAssoc(op)
         val sel = atPos(pos)(Select(stripParens(t), op.encode))
         if (targs.isEmpty) sel
-        else atPos(pos union targs.last.pos withPoint pos.point) {
-          TypeApply(sel, targs)
+        else {
+          /* if it's right-associative, `targs` are between `op` and `t` so make the pos transparent */
+          atPos((pos ^| targs.last.pos) makeTransparentIf !treeInfo.isLeftAssoc(op)) {
+            TypeApply(sel, targs)
+          }
```

Comment:
For my own reference in review:

```scala
  def |(that: Position): Position                   = this union that
  def ^(point: Int): Position                       = this withPoint point
  def |^(that: Position): Position                  = (this | that) ^ that.point
  def ^|(that: Position): Position                  = (this | that) ^ this.point
```

Reply:
I don't "have it in for punctuation, so these were pleasantly expressive to me. If you (of all people!) need to have a cross-reference handy, I'd say that's a sign that they may be less transparent than I had hoped. English names have been restored.